com.reprezen.swagedit/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessor.java

```diff
@@ -48,15 +51,36 @@
  * This class provides basic content assist based on keywords used by the
  * swagger schema.
  */
-public class SwaggerContentAssistProcessor extends TemplateCompletionProcessor implements IContentAssistProcessor {
+public class SwaggerContentAssistProcessor extends TemplateCompletionProcessor implements IContentAssistProcessor, ICompletionListener {
 
 	private final SwaggerProposalProvider proposalProvider = new SwaggerProposalProvider();
+	private final JsonReferenceProposalProvider referenceProposalProvider = new JsonReferenceProposalProvider();
+	private final ContentAssistant contentAssistant;
+
 	private String currentPath = null;
+	private int cyclePosition = 0;
```

Comment:
It would be helpful to add a javadoc or comment that it corresponds to `com.reprezen.swagedit.assist.JsonReferenceProposalProvider.Scope`